templates/pwm.asg.element.json

```
@@ -0,0 +1,1057 @@
+								"content": {
+									"Fn::Join": [
+										"",
+										[
+											"*/8 * * * * root /usr/local/bin/adminusermgmt.sh -G $(cat /usr/local/bin/admingroupname)",
+											"\n"
+										]
+									]
+								},
+								"group": "root",
+								"mode": "000644",
+								"owner": "root"
+							},
+							"/usr/local/bin/admingroupname": {
+								"content": {
+									"Ref": "AdminIAMGroup"
+								},
+								"group": "root",
+								"mode": "000600",
+								"owner": "root"
+							},
+							"/usr/local/bin/authorized_keys_command.sh": {
+								"content": {
+									"Fn::Join": [
+										"",
+										[
+											"#!/bin/bash -e",
+											"\n",
+											"if [ -z \"$1\" ]; then",
+											"\n",
+											"  exit 1",
+											"\n",
+											"fi",
+											"\n",
+											"aws iam list-ssh-public-keys --user-name \"$1\" --query \"SSHPublicKeys[?Status == 'Active'].[SSHPublicKeyId]\" --output text | while read KeyId; do",
+											"\n",
+											"  aws iam get-ssh-public-key --user-name \"$1\" --ssh-public-key-id \"$KeyId\" --encoding SSH --query \"SSHPublicKey.SSHPublicKeyBody\" --output text",
+											"\n",
+											"done",
+											"\n"
+										]
+									]
+								},
+								"group": "root",
+								"mode": "000755",
+								"owner": "root"
+							}
+						}
+					},
+					"d-discousersetup": {
+						"commands": {
+							"10-cp_discosshusermgmt": {
+								"command": {
+									"Fn::Join": [
+										"",
+										[
+											"cp /srv/salt/pwm/discousermgmt.sh /usr/local/bin/discousermgmt.sh",
+											"\n"
+										]
+									]
+								}
+							},
+							"20-ch_discosshusermgmt": {
+								"command": {
+									"Fn::Join": [
+										"",
+										[
+											"chmod 700 /usr/local/bin/discousermgmt.sh & chown root /usr/local/bin/discousermgmt.sh & chgrp root /usr/local/bin/discousermgmt.sh",
+											"\n"
+										]
+									]
+								}
+							},
+							"30-rundiscosshusermgmt": {
+								"command": {
+									"Fn::Join": [
+										"",
+										[
+											"/usr/local/bin/discousermgmt.sh -G $(cat /usr/local/bin/discogroupname)",
+											"\n"
+										]
+									]
+								}
+							}
+						},
+						"files": {
+							"/etc/cron.d/discousermgmt": {
+								"content": {
+									"Fn::Join": [
+										"",
+										[
+											"*/8 * * * * root /usr/local/bin/discousermgmt.sh -G $(cat /usr/local/bin/discogroupname)",
+											"\n"
+										]
+									]
+								},
+								"group": "root",
+								"mode": "000644",
+								"owner": "root"
+							},
+							"/usr/local/bin/discogroupname": {
+								"content": {
+									"Ref": "DiscoIAMGroup"
+								},
+								"group": "root",
+								"mode": "000600",
+								"owner": "root"
+							}
+						}
+					},
+					"e-cloudwatchagentsetup": {
+						"commands": {
+							"10-get-agent": {
+								"command": {
```

All these one-line commands with no references can be simplified significantly:
```suggestion
								"command": "wget https://s3.amazonaws.com/amazoncloudwatch-agent/linux/amd64/latest/AmazonCloudWatchAgent.zip"
```